src/api/c++/z3++.h

```diff
@@ -559,32 +643,30 @@ namespace z3 {
         }
         */
 
-        class iterator {
+        class iterator final {
```

Comment:
Since `iterator` will not be extended, we can declare it `final`. This has optimization benefits.